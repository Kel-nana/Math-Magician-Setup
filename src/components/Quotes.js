import React, { useState, useEffect } from 'react';

export default function Quotes() {
  // API URL and token
  const apiUrl = 'https://api.api-ninjas.com/v1/quotes/';
  const token = '7R4Z2M0ATFoFWAqNgcKVig==auatjCNk3a95D2M7';

  // State variables
  const [quote, setQuote] = useState(''); // current quote
  const [isLoading, setIsLoading] = useState(true); // loading indicator
  const [error, setError] = useState(''); // error message

  // Fetch quote on mount
  useEffect(() => {
    fetch(apiUrl, {
      headers: { 'X-API-Key': `${token}` },
    })
      .then((res) => {
        if (!res.ok) {
          throw new Error('An error occurred while fetching the quote.');
        }
        return res.json();
      })
      .then((data) => {
        setQuote(data[0].quote); // update state with fetched quote
        setIsLoading(false); // set loading indicator to false
      })
      .catch((err) => {
        setError(err.message); // set error message
        setIsLoading(false); // set loading indicator to false
        throw new Error(err);
      });
  }, []); // empty dependency array to run once on mount

  // Render quote or loading/error message
  return (
    <div className="quote-bodybg">
      <div className="quote-body">
        <div className="quote">
          <span>{isLoading ? 'Loading ...' : ''}</span>
          <span>{error || ''}</span>
          {quote}
        </div>
      </div>
    </div>
  );
}
