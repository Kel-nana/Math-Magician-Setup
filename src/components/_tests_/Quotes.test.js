import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Quotes from '../Quotes';

describe('Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Quotes />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
