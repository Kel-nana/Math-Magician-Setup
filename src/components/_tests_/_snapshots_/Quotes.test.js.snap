// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
  <span />
  <span />
</section>
 <div className="quote-bodybg">
      <div className="quote-body">
        <div className="quote">
          <span>Loading ...</span>
          <span>An error occurred while fetching the quote.</span>
          
        </div>
      </div>
    </div>
`;